data-prepper-plugins/dynamodb-source/README.md

```diff
@@ -0,0 +1,83 @@
+# DynamoDB Source
+
+This is a source plugin that supports retrieve data from DynamoDB tables. Basic use case of this source plugin is to
+sync the data from DynamoDB tables to OpenSearch indexes. With this CDC support, customer can run the end to end data
+sync pipeline and capture changed data in near real-time without writing any codes and without any downtime of business.
+Such pipeline can run on multiple nodes in parallel to support data capture of large scale tables.
+
+This plugin can support below three different modes:
+
+1. Full load only:  One time full data export and load
+2. CDC Only:  DynamoDB Stream
+3. Full load + CDC:  One time full export and load + DynamoDB Stream.
+
+## Usages
+
+To get started with this DynamoDB source, create the following source configuration:
+
+```yaml
+source:
+  dynamodb:
+    tables:
+      - table_arn: "arn:aws:dynamodb:us-west-2:123456789012:table/my-table"
+        stream:
+          start_position:
+        export:
+          s3_bucket: "my-bucket"
+          s3_prefix: "export/"
+    aws:
+      region: "us-west-2"
+      sts_role_arn: "arn:aws:iam::123456789012:role/DataPrepperRole"
+
+    coordinator:
+      dynamodb:
+        table_name: "coordinator-demo"
+        region: "us-west-2"
```

Comment:
Sure. Will do once this code is merged.  I will create one to refer to this.

It looks like we can either expose a getCoordinationStore from current LeaseBasedCoordinator or by default only create coordinationStore from the data-prepper-config.yaml  only instead of  the coordinator. Not sure what is the best way.

Reply:
We will want to use the data-prepper-config only. We can use the same flow that we currently use to provide plugins with a source coordinaor